### PR TITLE
Refactor/build system

### PR DESCRIPTION
**Pull request by mtoso**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.DS_Store
```
Comment (ThePrimeagen) [inline .gitignore:1]: osx?  Chump

File: examples/ws-server/client.node.ts [hunk 2]
```diff
@@ -2,7 +2,6 @@ import {
     NetworkPipe,
 } from '../../src/types';
 
-import Platform from '../../src/Platform';
```
Comment (ThePrimeagen) [inline examples/ws-server/client.node.ts:5]: is client.node.ts unable to run?  Should I simply import ../../src/node/Platform?

Comment (mtoso) [reply examples/ws-server/client.node.ts:5]: I think I forgot to change this import. you can do import Platform from "../../#{platform}/Platform"

Comment (mtoso) [reply examples/ws-server/client.node.ts:5]: sorry its #{target} not #{platform}

**Pull request merged**